### ISSUE TITLE
1.18.2 startup crash - "mixins.shetiphiancore.json:SPC_AdaptiveAO failed injection check, (0/1) succeeded."

### ISSUE THREAD

**Issue by xFirefalconx** [ShetiPhian/Localizations-and-Issues#434]
**Describe the bug**
Game crashes on startup in my modpack (devving new pack version rn). Doesn't happen without your mod(s) (platforms and the core mod).
I hopefully found the correct line in my crash log, not a genius with it tbh. 
If you need more info, let me know and I will *some day* do a crash analysis run but since that might take 2-3 weeks in itself (just the test runs) I hope you can take some info from the data I can provide right now. There's no time pressure or anything but less time consuming work is always beneficial. Thanks in advance for taking a look. If you got a candidate down who is actually at fault if it's not the core mod, let me know please so I can see whether the other dev(s) involved can do something about it or not. 

**Additional context**
Minecraft Version: 1.18.2
Forge Version: 40.2.17
ShetiPhian Core Version: 3.10.17
My Mod(s) Name & Version: (platforms-forge-1.18.2-1.9.10) actually, just the core mod is enough to make it crash. Just tested. 
Are other mods present?
If yes:
 ... haa here we go again, I'll give just the list file: https://gist.github.com/xFirefalconx/50fffabcf7c1adb257e61df37def7655 I have a sneaking suspicion it might be Canary, or Modernfix, or Redirector or something akin to that. Just a hunch though I couldn't test it yet. 
- [x] - Does everything work without my mods? - yup, the game loads and the worlds are playable

- [x] - Have you tested with only my mods? - loads the game without interference. 

PasteBin or GitHub Gists links
latest.log = https://gist.github.com/xFirefalconx/f1f906ab72ed021fc61e85501987e83a
crash.log = https://gist.github.com/xFirefalconx/535a71a94a883a64863d935194643472

**Comment by ArchaicDeity** [comment]
Hi, Did you ever solve the issue?

**Comment by xFirefalconx** [comment]
Nope, sadly I had other issues I had to attend to so I couldn't further dabble in it. I found CTM mod to be crashing a lot though so I might try test if the issue still persists soon (TM). Dunno if it was active back when I tried.

**Comment by ArchaicDeity** [comment]
Ohhh okay thanks I found a solution before but I took a few steps back with my modpack and now I'm stuck with the problem again and I can't remember what I did last time.

**Comment by ArchaicDeity** [comment]
Hi again...Can I ask what version of Shetiphiancore are you using I can't find it in your mods list.

**Comment by xFirefalconx** [comment]
Hi, thanks for your input. Mine is indeed 1.18.2-3.10.17, will try see what downgrading does. Which version does Platforms need to be compatible with 3.10.16? 

It very likely is a third party. I had one error in my modpack editing that was randomly appearing on a specific combination of events in-game. It took me days after days after slowly going insane to find the mod causing it. It might be such an issue here as well. I'll post again once I had time to test and also add my latest modlist to that post. Let's hope that downgrade does the trick for now.

**Comment by xFirefalconx** [comment]
Can confirm, the downgrade works for my modpack. 
Modlist (haven't tried platforms yet so it's void in the list, I copy it ingame):
https://gist.github.com/xFirefalconx/d21fc74d86cfc75a433048532d11d0ec

**Comment by xFirefalconx** [comment]
All right, thanks. I'll test the 1.9.9 then. I'll also post any things I notice when going back to debug mine further. 

Very true. I have had people ask me how that amount of mods even runs without hassle, haha. I mainly group my pack into performance and enhancement mods. I think about 25% (more or less) are solely to enhance the game's performance. I usually note each mod in my changelogs that I implement in the modpack file system as text files, on my github where I keep track of changes and also in the uploaded version list. I also write the reason for adding it behind each single of these 600+ mods... took a bit of time. Most mods don't do extremely much, some just add one flower or a block type. Some use a better culling technique, I also got analytics mods like Spark in there, loggers, etc. The main core around this is using ModernFix and its recommended mod list, FPSreducer for standby actions to not kill your system when tabbing out, MemoryLeakFix, and others of that category. Best is to add them one by one and test between adding the next mod. If you put in "Fix" in the search bar of your preferred distribution site, it usually shows a lot of nice mods. Also FPS, Performant, Lag, Entity, Culling. Other than that, read a bit through community discussions on large modding servers, they might also give away tips to fix issues. 

My allocated RAM also mixes in with the performance, I had a lot of lag spikes on 9600MB (which already seems like a lot) and MemorySweep had to freeze my system every x minutes in order to clean a bit of RAM, but on 11008 MB it runs quite smoothly right now - I might have to allocate more later though as I haven't done extreme situation testing yet. I am using virtual RAM on my HDD drive on top of using 24GB DDR3 RAM (2x4GB, 2x8GB (I'm gonna upgrade this though... poor hard drive)) on a Windows 10 machine (this is whack but it works : D). My CPU is a i5-4690K, an older model and definitely a potato by current measures. And a GTX 1060 6GB VRAM.
 
My FPS, depending on shader and render settings, are between an average of 11-15FPS in standby (tabbed out/pause screen) and an average of 21-37FPS (spikes up to 47 FPS and higher depending on the environment) during play time though those measures are from my non-crowded singleplayer world in petrified mode (a pack playmode), I'd have to also test the wilds version (which is more vanilla style) and observe over a larger amount of time than a few random seconds of glancing at the FPS counter on-screen. It's still in development so the numbers may vary but I'd assume it's pretty stable, given potato pc with a bit of extra RAM, High fidelity shaders and the current stage of testing.

BYG didn't cause such lag spikes for me yet, though I can't fully say what cause it could have in your pack. I use version 1.4.7 right now (might update though depending on what the changelog says). A little bit of frame reduction might though be valid, given they add a lot of blocks and things. Have you tried testing whether that issue stays with BYG dynamic trees addon? It might be an indicator if the issue is gone when replacing the trees or sorts. Hope that info helps you somehow!

**Comment by ArchaicDeity** [comment]
Okay, I'll update as well if I find the actual culprit.

I really wish I was nearly as organized as you are that would definitely have saved me a lot of headaches and time...That's something I've been wanting to get around to making sure each mod has a good reason for being in the pack aswell as organizing them. Modernfix is definitely a safer I am aiming to get more performance for my pack first before using it my thinking process behind this is if I can get atleast good performance without it the performance with it would be amazing, adding them one by one is a mistake I've made few to many times getting consumed in mods that look good and just going down a rabbit hole then the game crashes and it's a few hours of finding the culprit(I do think this has helped me a lot with reading logs and crash reports though).

I don't think that's too bad actually the memory allocated not the freezing your system...I also still need too test I'm thinking especially Create and just a lot of entities running at one for Mob farms. I still have an older pc hear I'm thinking of using for testing actually with an i5-2500 and then I have an extra gt 1080 I think if I can get my pack to run on that it will be safe to run on mostly anything.

I actually think 20fps is playing if it stays constant before I had my current setup I played and completed Spider-Man Remastered with a 25fps average(running off integrated graphics)...Petrified mode seems interesting, I think any pack this size takes years to develop and a lot of playtesting I usually join a world and just totally try to kill fps(using a repeating command block to summon a bunch of dragons or creating something big with create like upping the limit of the blocks that can be mopve and moving 20000 blocks with a piston) I think having it stable is more important than higher fps because getting 60fps and having it randomly drop to 15fps is just annoying. 

I'll test it out I removed it pretty early on as I was having random drops and I think it was a reddit post or something talking about BYG causing it and well I haven't had the same problem.

Have you tried pregen for worlds?
What are your launch times loading into main menu and then into world?

I don't know if you use reddit this is a post I made about 6 months back about progress with my pack.
Note: An extreme amount of progress has been made since then load times have been improved by a lot as well as fps.

[https://www.reddit.com/r/feedthebeast/comments/16n1e3y/640_mods_on_45gb_of_ram_1182aswell_as_a_cry_for/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button](url)
